snapmaker/src/service/system.cpp

```diff
@@ -340,7 +340,7 @@ ErrCode SystemService::ResumeTrigger(TriggerSource source) {
 
   switch (ModuleBase::toolhead()) {
   case MODULE_TOOLHEAD_3DP:
-    if (runout.is_filament_runout()) {
+    if (printer1->GetFilamentState()) {
```

Comment:
recommend to call printer1->GetFilamentState() in the runout.is_filament_runout()